tests/credential_plugins_test.py

```diff
@@ -131,6 +132,67 @@ def test_hashivault_handle_auth_not_enough_args():
         hashivault.handle_auth()
 
 
+def test_aws_assumerole_with_accesssecret():
+    kwargs = {
+        'access_key': 'my_access_key',
+        'secret_key': 'my_secret_key',
+        'role_arn': 'the_arn',
+        'identifier': 'access_token',
+    }
+    with mock.patch.object(aws_assumerole, 'aws_assumerole_getcreds') as method_mock:
```

Comment:
In `pytest,` it's customary to use `monkeypatch`. If `mock` is needed, it should be integrated through the `pytest-mock` plugin and injected as a `mocker` fixture.